src/middlewared/middlewared/plugins/webui/crypto.py

```diff
@@ -0,0 +1,22 @@
+from middlewared.schema import accepts, Int
+from middlewared.service import Service
+
+
+class WebUICryptoService(Service):
+
+    class Config:
+        namespace = 'webui.crypto'
+        private = True
+        cli_private = True
+
+    @accepts(roles=['READONLY'])
```

Comment:
This is great! However, let's add 2 new roles `CERT_READ` and `CERT_WRITE`. Then you can add `CERT_READ` as a member of `READONLY`. This follows same paradigm as the other plugins we've done.